packages/@react-native-windows/telemetry/src/telemetry.ts

```diff
@@ -4,207 +4,411 @@
-  if (parens !== -1) {
-    // case 1: method === 'methodName (rootOfThePath'
-    frame.method = frame.method.substr(0, parens).trim();
-  } else {
-    // case 2: method === <no_method> or something without '(', fileName is full path
-  }
-  // preserve only the last_directory/filename
-  frame.fileName = path.join(
-    path.basename(path.dirname(frame.fileName)),
-    path.basename(frame.fileName),
-  );
-  frame.assembly = '';
-}
+  private static trackCommandEvent(extraProps?: Record<string, any>) {
+    const props: Record<string, any> = {
+      eventName: 'RNWCLI.Command',
+    };
 
-export function tryGetErrorCode(msg: string): string | undefined {
-  const errorRegEx = /error (\w+\d+):/gi;
-  const m = errorRegEx.exec(msg);
-  return m ? m[1] : undefined;
-}
+    // Set command props
+    props.command = {
+      options: Telemetry.commandInfo.startInfo?.options,
+      defaultOptions: Telemetry.commandInfo.startInfo?.defaultOptions,
+      args: Telemetry.commandInfo.startInfo?.args,
+      durationInSecs:
+        (Telemetry.commandInfo.endTime! - Telemetry.commandInfo.startTime!) /
+        1000,
+      resultCode: Telemetry.commandInfo.endInfo?.resultCode,
+    };
 
-/**
- * Remove PII from exceptions' stack traces and messages
- * @param envelope the telemetry envelope. Provided by AppInsights.
- */
-export function sanitizeEnvelope(envelope: any /*context: any*/): boolean {
-  if (envelope.data.baseType === 'ExceptionData') {
-    const data = envelope.data.baseData;
-    for (const exception of data.exceptions || []) {
-      for (const frame of exception.parsedStack) {
-        sanitizeFrame(frame);
-      }
-      const errorCode = tryGetErrorCode(exception.message);
-      data.properties.errorCode = errorCode;
-      // CodedError has non-PII information in its 'type' member, plus optionally some more info in its 'data'.
-      // The message may contain PII information. This can be sanitized, but for now delete it.
-      // Note that the type of data.exceptions[0] is always going to be ExceptionDetails. It is not the original thrown exception.
-      // https://github.com/microsoft/ApplicationInsights-node.js/issues/707
-      if (Telemetry.preserveMessages) {
-        exception.message = sanitizeMessage(exception.message);
-      } else {
-        delete exception.message;
+    // Set remaining common props
+    Object.assign(props, extraProps);
+    props.project = Telemetry.projectProp;
+    props.versions = Telemetry.versionsProp;
+
+    // Fire event
+    Telemetry.client!.trackEvent({name: props.eventName, properties: props});
+    Telemetry.client!.flush();
+  }
+
+  static trackException(error: Error, extraProps?: Record<string, any>) {
+    if (!Telemetry.client) {
+      return;
+    }
+
+    const props: Record<string, any> = {
+      eventName: 'RNWCLI.CodedError',
+    };
+
+    // Save off CodedError info
+    const codedError =
+      error instanceof errorUtils.CodedError
```

Comment:
Should we handle non-coded exceptions with a different event?

Reply:
I don't think it's necessary, our coded events are just superset of error with a symbolic error type. So we just pass up the type "Unknown" if a regular Error is thrown.